src/Bridges/bridgeoptimizer.jl

```diff
@@ -227,32 +227,60 @@ function MOI.get(b::AbstractBridgeOptimizer,
                  attr::MOI.AbstractConstraintAttribute,
                  ci::CI)
     if is_bridged(b, typeof(ci))
-        if MOI.is_set_by_optimize(attr)
-            MOI.get(b, attr, bridge(b, ci))
-        else
+        if attr isa MOI.ConstraintName || attr isa MOI.ConstraintFunction ||
+                attr isa MOI.ConstraintSet
             MOI.get(b.bridged, attr, ci)
+        else
+            MOI.get(b, attr, bridge(b, ci))
         end
     else
         MOI.get(b.model, attr, ci)
     end
 end
-## Setting names
-function MOI.supports(b::AbstractBridgeOptimizer, attr::MOI.ConstraintName,
-                      Index::Type{<:CI})
-    if is_bridged(b, Index)
-        return MOI.supports(b.bridged, attr, Index)
+function MOI.supports(b::AbstractBridgeOptimizer,
+                      attr::MOI.AbstractConstraintAttribute,
+                      IndexType::Type{CI{F, S}}) where {F,S}
+    if is_bridged(b, IndexType)
+        if attr isa MOI.ConstraintName
+            return MOI.supports(b.bridged, attr, IndexType)
+        else
+            return MOI.supports(b, attr, concrete_bridge_type(b, F, S))
+        end
     else
-        return MOI.supports(b.model, attr, Index)
+        return MOI.supports(b.model, attr, IndexType)
     end
 end
-function MOI.set(b::AbstractBridgeOptimizer, attr::MOI.ConstraintName,
-                  constraint_index::CI, name::String)
-    if is_bridged(b, typeof(constraint_index))
-        MOI.set(b.bridged, attr, constraint_index, name)
+
+function MOI.set(b::AbstractBridgeOptimizer,
+                 attr::MOI.AbstractConstraintAttribute,
+                 index::CI, value)
+    if is_bridged(b, typeof(index))
+        if attr isa MOI.ConstraintName
+            return MOI.set(b.bridged, attr, index, value)
+        else
+            return MOI.set(b, attr, bridge(b, index), value)
+        end
     else
-        MOI.set(b.model, attr, constraint_index, name)
+        return MOI.set(b.model, attr, index, value)
     end
 end
+## Setting names
+# function MOI.supports(b::AbstractBridgeOptimizer, attr::MOI.ConstraintName,
+#                       Index::Type{<:CI})
+#     if is_bridged(b, Index)
+#         return MOI.supports(b.bridged, attr, Index)
+#     else
+#         return MOI.supports(b.model, attr, Index)
+#     end
+# end
+# function MOI.set(b::AbstractBridgeOptimizer, attr::MOI.ConstraintName,
```

Comment:
It might be more efficient to re-enable these methods and letting multiple dispatch do the work instead of using `isa`. in the generic `set` and `supports`